feat: add support for conformance testing 'typed' declarative function signature

garethgeorge opened the pull request:
To be used for conformance testing new `typed` declarative function signatures. 

Ruby example command:
```
./client -cmd="bundle exec functions-framework-ruby --source test/conformance/app.rb --target typed_func --signature-type http" --type=http --declarative-type=typed -buildpacks=false
```

In the case of typed functions, a JSON request is generated and it is asserted that the function returns a JSON object that echo's the request in the `payload` field of a new JSON object. I.e. for input `{"a": "b"}` the function should return `{"payload": {"a": "b"}}`.

kenneth-rosario commented:
Can you comment an example of how the test functions would look like in any one of the supported languages of your choice for reference?

garethgeorge commented:
The test function we'll add for ruby is:

```
FunctionsFramework.typed "typed_func" do |request|
  return {
    :payload => request,
  }
end
```

kenneth-rosario reviewed (commented):
Lets update the README to include new flag as well.

garethgeorge commented:
Thanks for the initial review Kenneth! Went ahead and updated README.md with examples for the new conformance test type. 

I think I just need another (final this time!) pass to cover the files I've added to update the github actions workflows.

kenneth-rosario commented on README.md:
````diff
@@ -58,6 +58,15 @@ Frameworks to the Functions Framework contract.
+          --type=http \
+          --declarative-type=typed \
+          -buildpacks=false
+        ```
````
Lets update [the usage table](https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/README.md#usage) and step 1 of the [quick start] (https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/README.md#quickstart)

garethgeorge commented:
Done, updated flags list.

kenneth-rosario reviewed (approved):
This looks great!
